pkg/controller/nodes/task/handler.go

```diff
@@ -133,6 +143,7 @@ type Handler struct {
 	catalog         catalog.Client
 	asyncCatalog    catalog.AsyncClient
 	plugins         map[pluginCore.TaskType]pluginCore.Plugin
+	taskMetricsMap  map[pluginCore.TaskType]*taskMetrics
```

Comment:
```suggestion
	taskMetricsMap  map[string]*taskMetrics
```

or a new `MetricKey` type alias?